src/main/java/org/openstreetmap/atlas/geography/atlas/delta/AtlasDeltaGenerator.java

```diff
@@ -44,55 +70,134 @@ protected int onRun(final CommandMap command)
+    }
+
+    private void compareShardByShard(final Path before, final Path after,
+            final Path outputDirectory)
+    {
+        final List<File> afterShardFiles = fetchAtlasFilesInDirectory(after);
+        afterShardFiles.parallelStream().forEach(afterShardFile ->
+        {
+            final Path beforeShardPath = before.resolve(afterShardFile.getName());
+            final Atlas beforeAtlas = loadSingleAtlas(beforeShardPath);
+            final Atlas afterAtlas = new AtlasResourceLoader().load(afterShardFile);
+            compare(beforeAtlas, afterAtlas, outputDirectory);
+        });
+    }
+
+    private void compare(final Atlas beforeAtlas, final Atlas afterAtlas,
+            final Path outputDirectory)
+    {
+        final String name = FilenameUtils.removeExtension(beforeAtlas.getName());
 
         final AtlasDelta delta = new AtlasDelta(beforeAtlas, afterAtlas).generate();
 
         final String text = delta.toDiffViewFriendlyString();
         final File textFile = new File(
-                outputDir.resolve(name + FileSuffix.TEXT.toString()).toFile());
+                outputDirectory.resolve(name + FileSuffix.TEXT.toString()).toFile());
         textFile.writeAndClose(text);
-        this.logger.info("Saved txt file {}", textFile);
+        this.logger.info("Saved text file {}", textFile);
 
         final String geoJson = delta.toGeoJson();
         final File geoJsonFile = new File(
-                outputDir.resolve(name + FileSuffix.GEO_JSON.toString()).toFile());
+                outputDirectory.resolve(name + FileSuffix.GEO_JSON.toString()).toFile());
         geoJsonFile.writeAndClose(geoJson);
         this.logger.info("Saved GeoJSON file {}", geoJsonFile);
 
         final String relationsGeoJson = delta.toRelationsGeoJson();
         final String relationsGeoJsonFileName = name + "_relations"
                 + FileSuffix.GEO_JSON.toString();
         final File relationsGeoJsonFile = new File(
-                outputDir.resolve(relationsGeoJsonFileName).toFile());
+                outputDirectory.resolve(relationsGeoJsonFileName).toFile());
         relationsGeoJsonFile.writeAndClose(relationsGeoJson);
         this.logger.info("Saved Relations GeoJSON file {}", relationsGeoJsonFile);
     }
 
-    private Atlas load(final Path path)
-    {
-        final File file = new File(path.toFile());
-        return new AtlasResourceLoader().load(file);
-    }
-
-    private void run(final Path before, final Path after, final Path outputDir)
+    private static List<File> fetchAtlasFilesInDirectory(final Path directory)
     {
-        this.logger.info("Comparing {} and {}", before, after);
-        final Atlas beforeAtlas = load(before);
-        final Atlas afterAtlas = load(after);
-        compare(beforeAtlas, afterAtlas, outputDir);
+        return new File(directory.toFile()).listFilesRecursively().stream()
+                .filter(file -> "atlas".equals(FilenameUtils.getExtension(file.getName())))
```

Comment:
Recommend reusing the `AtlasResourceLoader` `IS_ATLAS` predicate here, this will also allow for gzipped files to be compared. [Reference](https://github.com/osmlab/atlas/blob/053eed9d3520b5791faafbd8af1aeb9ca33806ca/src/main/java/org/openstreetmap/atlas/geography/atlas/AtlasResourceLoader.java#L57).

If that doesn't work, let's rely on `FileSuffix` instead of a raw string.

Reply:
Cool, will try it out!